src/main/java/com/google/api/codegen/transformer/py/PythonPackageMetadataTransformer.java

```diff
@@ -0,0 +1,182 @@
+    List<ViewModel> topLevelMetadata =
+        Lists.transform(
+            getTopLevelTemplateFileNames(),
+            new Function<String, ViewModel>() {
+              @Override
+              @Nullable
+              public ViewModel apply(@Nullable String templateFileName) {
+                return generateMetadataView(model, apiConfig, templateFileName);
+              }
+            });
+
+    List<ViewModel> metadata = new ArrayList<>();
+    metadata.addAll(initFiles);
+    metadata.addAll(topLevelMetadata);
+    return metadata;
+  }
+
+  @Override
+  public List<String> getTemplateFileNames() {
+    List<String> templates = new ArrayList<>();
+    templates.addAll(getTopLevelTemplateFileNames());
+    templates.addAll(getInitTemplateFileNames());
+    return templates;
+  }
+
+  public List<String> getTopLevelTemplateFileNames() {
+    return Lists.newArrayList(
+        "LICENSE.snip",
+        "py/MANIFEST.in.snip",
+        "py/PUBLISHING.rst.snip",
+        "py/setup.py.snip",
+        "py/requirements.txt.snip",
+        "py/README.rst.snip",
+        "py/tox.ini.snip");
+  }
+
+  public List<String> getInitTemplateFileNames() {
+    return Lists.newArrayList("py/__init__.py.snip", "py/namespace__init__.py.snip");
+  }
+
+  private ViewModel generateMetadataView(Model model, ApiConfig apiConfig, String template) {
+    int extensionIndex = template.lastIndexOf(".");
+    String outputPath = template.substring(0, extensionIndex);
+
+    return PackageMetadataView.newBuilder()
+        .templateFileName(template)
+        .outputPath(outputPath)
+        .identifier(apiConfig.getDomainLayerLocation())
+        .packageVersion(packageConfig.packageVersion("python"))
+        .protoPath(packageConfig.protoPath())
+        .shortName(packageConfig.shortName())
+        .gaxVersion(packageConfig.gaxVersion("python"))
+        .protoVersion(packageConfig.protoVersion("python"))
+        .commonProtosVersion(packageConfig.commonProtosVersion("python"))
+        .packageName(packageConfig.packageName("python"))
+        .majorVersion(packageConfig.apiVersion())
+        .author(packageConfig.author())
+        .email(packageConfig.email())
+        .homepage(packageConfig.homepage())
+        .license(packageConfig.license())
+        .fullName(model.getServiceConfig().getTitle())
+        .serviceName(apiConfig.getPackageName())
+        .namespacePackages(
+            computeNamespacePackages(apiConfig.getPackageName(), packageConfig.apiVersion()))
+        .build();
+  }
+
+  private List<String> computePackages(String packageName) {
+    ArrayList<String> packages = new ArrayList<>();
+    StringBuilder current = new StringBuilder();
+    boolean first = true;
+    for (String pkg : Splitter.on(".").split(packageName)) {
+      if (!first) {
+        current.append("." + pkg);
+      } else {
+        current.append(pkg);
+        first = false;
+      }
+      packages.add(current.toString());
+    }
```

Comment:
A Python package name `foo.bar.baz` implicitly defines three packages: `foo`, `foo.bar`, and `foo.bar.baz`. Is there a more concise way to express this in Java?